src/app/components/add-course/authors/authors.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit, forwardRef } from '@angular/core';
+import { ControlValueAccessor, Validator, AbstractControl, ValidationErrors, NG_VALUE_ACCESSOR, NG_VALIDATORS } from '@angular/forms';
+import { HttpClient } from '@angular/common/http';
+import { Author, AuthorRaw } from 'interfaces/authors';
+
+const authorsEndpoint = 'http://localhost:3004/authors';
+
+@Component({
+   selector: 'app-authors',
+   templateUrl: './authors.component.html',
+   styleUrls: ['./authors.component.styl'],
+   providers: [
+      {
+         provide: NG_VALUE_ACCESSOR,
+         useExisting: forwardRef(() => AuthorsComponent),
+         multi: true
+      },
+      {
+         provide: NG_VALIDATORS,
+         useExisting: AuthorsComponent,
+         multi: true
+      }
+   ]
+})
+export class AuthorsComponent implements OnInit, ControlValueAccessor {
+
+   authors: string[] = [];
+   authorsFromServer: string[] = [];
+   authorsInputValue = '';
+
+   constructor(private http: HttpClient) { }
+
+   ngOnInit() {
+   }
+
+   validate(control: AbstractControl): ValidationErrors | null {
+      if (!control.value || !control.value.length) {
+         return {
+            duration: {
+               valid: false
+            }
+         };
+      }
+      return null;
+   }
+
+   writeValue(authors: AuthorRaw[]) {
+      if (authors === null) {
+         this.authors = [];
+      } else {
+         const processedAuthors = authors.map(author => `${author.firstName} ${author.lastName}`);
+         this.authors = this.authors.concat(processedAuthors);
+      }
+   }
+
+   onChange: any = () => {};
+
+   registerOnChange(callback) {
+      this.onChange = callback;
+   }
+
+   registerOnTouched() {
```

Comment:
Delete methods, if you don't use them.

Reply:
@JuliaAlaeva this is an obligatory method for interface ControlValueAccessor